### PR TITLE
Fix misc bugs, h/t Psalm

### PR DESCRIPTION
**Pull request by mhsdef**
Lil' here, lil' there.

### CODE REVIEW

File: inc/Editor/DataBinding/BlockBindings.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ private static function apply_query_input_overrides( array $query_input, array $
 			$block_name
 		);
 
-		return array_merge( $query_input, $query_input_overrides );
+		return array_merge( $query_input, $overrides );
```
Comment (mhsdef) [inline inc/Editor/DataBinding/BlockBindings.php:152]: I'm not sure offhand how to easily test, but, I'm assuming this was the intent. 

This one look right to you, @chriszarate?

File: inc/Config/QueryContext/HttpQueryContext.php [hunk 2]
```diff
@@ -24,35 +24,30 @@ class HttpQueryContext implements QueryContextInterface, HttpQueryContextInterfa
...
-	 *       @type array {
-	 *         @type $target Targeted override.
-	 *         @type $type   Override type.
-	 *       }
-	 *     }
-	 *     @type string $type         The variable type (string, number, boolean)
-	 *   }
-	 * }
-	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingTraversableTypeHintSpecification
+	 * The variable type (string, number, boolean) for both cases of type.
```
Comment (chriszarate) [inline inc/Config/QueryContext/HttpQueryContext.php]: This looks messed up, plus we lost comments

File: inc/Config/QueryContext/HttpQueryContext.php [hunk 3]
```diff
@@ -24,35 +24,30 @@ class HttpQueryContext implements QueryContextInterface, HttpQueryContextInterfa
...
-	 *   @type array $var_name {
-	 *     @type string $default_value Optional default value of the variable.
-	 *     @type string $name          Display name of the variable.
-	 *     @type string $path          JSONPath expression to find the variable value.
-	 *     @type string $type          The variable type (string, number, boolean)
-	 *   }
-	 * }
-	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingTraversableTypeHintSpecification
+	 * 
+	 * Uses JSONPath for path, the variable type (string, number, boolean) for type.
```
Comment (chriszarate) [inline inc/Config/QueryContext/HttpQueryContext.php]: same here

File: inc/Integrations/VipBlockDataApi/VipBlockDataApi.php [hunk 4]
```diff
@@ -15,14 +15,21 @@ class VipBlockDataApi {
 
 	public static function init(): void {
 		add_filter( 'vip_block_data_api__sourced_block_result', [ __CLASS__, 'resolve_remote_data' ], 5, 4 );
-		add_filter( 'vip_block_data_api__after_parse_blocks', [ __CLASS__, 'add_debug_info' ], 10, 2 );
+		add_filter( 'vip_block_data_api__after_parse_blocks', [ __CLASS__, 'add_debug_info' ], 10, 1 );
```
Comment (chriszarate) [inline inc/Integrations/VipBlockDataApi/VipBlockDataApi.php]: Can we fix this the other way, by adding post_id to the debug output?

File: inc/REST/AuthController.php [hunk 5]
```diff
@@ -2,8 +2,6 @@
 
 namespace RemoteDataBlocks\REST;
 
-use WP_REST_Controller;
```
Comment (chriszarate) [inline inc/REST/AuthController.php:5]: Is this a code style thing or actually enforced by Psalm? I sort of prefer the use statements for consistency, but don't know how others feel

Comment (mhsdef) [reply inc/REST/AuthController.php:5]: It probably can be, but, It's not enforced anywhere. I have mixed feelings myself. On one hand, I aesthetically dislike the `\` but OTOH I feel like it communicates more clearly in down page context.

I'll put it back.

**Pull request merged**